File: benchmark/noinline_calls/calls.js
function mix(value, i) {
  return ((value * 17) + (i % 1009) + 23) % 1000003;
}

function main() {
  const iterations = 20000000;
  let value = 7;

  for (let i = 0; i < iterations; i++) {
    value = mix(value, i);
  }

  console.log(`Call sum: ${value}`);
}

main();

File: benchmark/noinline_calls/calls.py
def mix(value, i):
    return ((value * 17) + (i % 1009) + 23) % 1000003


def main():
    iterations = 20000000
    value = 7

    for i in range(iterations):
        value = mix(value, i)

    print(f"Call sum: {value}")


if __name__ == "__main__":
    main()
